src/test/java/org/apache/sling/event/impl/jobs/JobManagerImplTest.java

```diff
@@ -19,27 +19,162 @@
+		this.info = new QueueInfo();
+		info.queueConfiguration = mock(InternalQueueConfiguration.class);
+		when(queueConfigMgr.getQueueInfo(anyString())).thenReturn(info);
+		when(configuration.getQueueConfigurationManager()).thenReturn(queueConfigMgr);
+
+		this.capabilities = mock(TopologyCapabilities.class);
+		when(configuration.getTopologyCapabilities()).thenReturn(capabilities);
+
+		when(capabilities.detectTarget(eq("is/assigned"), any(), any())).thenReturn("assigned");
+
+		ResourceResolver resolver = mock(ResourceResolver.class);
+		when(resolver.getResource(anyString())).thenReturn(mock(Resource.class));
+		when(configuration.createResourceResolver()).thenReturn(resolver);
+
+		this.outContent = new ByteArrayOutputStream();
+		this.originalErr = System.err;
+		System.setErr(new PrintStream(outContent));
+
+	}
+
+	// SLING-8413
+	@Test
+	public void testTopicEscaping() {
+		String baseQuery = JobManagerImpl.buildBaseQuery(QUERY_ROOT, "randomNonQuotedTopic", QUERY_TYPE, false);
+		assertEquals(
+				"/jcr:root/var/eventing/foobar/element(*,slingevent:Job)[@event.job.topic = "
+						+ "'randomNonQuotedTopic' and not(@slingevent:finishedState) and @event.job.started.time",
+				baseQuery);
+
+		String baseQuery2 = JobManagerImpl.buildBaseQuery(QUERY_ROOT, "random'Topic", QUERY_TYPE, false);
+		assertEquals("/jcr:root/var/eventing/foobar/element(*,slingevent:Job)[@event.job.topic = "
+				+ "'random''Topic' and not(@slingevent:finishedState) and @event.job.started.time", baseQuery2);
+
+	}
+
+	@Test
+	public void wontLogAtWarn() throws IllegalAccessException, IOException, NoSuchMethodException, SecurityException,
+			IllegalArgumentException, InvocationTargetException {
+		JobManagerImpl jobManager = getJobManager(LEVEL_WARN);
+
+		jobManager.addJob("not/assigned", null, null);
+		assertFalse(logContainsLine(line -> line
+				.contains("Persisting job Sling Job [topic=not/assigned] into queue null with no assigned target")));
+
+		jobManager.addJob("is/assigned", null, null);
+		assertFalse(logContainsLine(line -> line
+				.contains("Persisting job Sling Job [topic=is/assigned] into queue null with no assigned target")));
+	}
+
+	@Test
+	public void logsInfoForUnassigned() throws IllegalAccessException, IOException, NoSuchMethodException,
+			SecurityException, IllegalArgumentException, InvocationTargetException {
+		JobManagerImpl jobManager = getJobManager(LEVEL_INFO);
+		jobManager.addJob("not/assigned", null, null);
+		assertTrue(logContainsLine(line -> line
+				.contains("Persisting job Sling Job [topic=not/assigned] into queue null with no assigned target")));
+
+		jobManager.addJob("is/assigned", null, null);
+		assertTrue(logContainsLine(line -> !line
+				.contains("Persisting job Sling Job [topic=is/assigned] into queue null with no assigned target")));
+	}
+
+	@Test
+	public void logsDebugForAssigned() throws IllegalAccessException, IOException, NoSuchMethodException,
```

Comment:
For the sake of completeness, I think we should test the condition where `target` is not assigned,  just like we did for the `logsInfoForUnassigned()` & `wontLogAtWarn()` test cases.

Reply:
At that point we're testing SLF4J functionality, which seems excessive.